File: backend/src/main/java/sn/immosn/backend/signalement/service/SignalementService.java
package sn.immosn.backend.signalement.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import sn.immosn.backend.client.web.signalement.dto.*;
import sn.immosn.backend.signalement.data.entity.StatutSignalement;

public interface SignalementService {
    SignalementResponseDto create(SignalementCreateRequestDto request, String clientEmail);
    Page<SignalementResponseDto> getClientSignalements(String clientEmail, StatutSignalement statut, Pageable pageable);
    Page<SignalementResponseDto> getAll(StatutSignalement statut, Pageable pageable);
    SignalementResponseDto getById(Long id);
    SignalementResponseDto updateStatut(Long id, UpdateStatutSignalementDto dto);
    void markAsRead(Long id);

    /** Historique des décisions d'un signalement (Sprint 4). */
    Page<SignalementHistoryDto> getHistory(Long id, Pageable pageable);
}

File: frontend/src/services/signalementService.js
import api from './api'

export default {
  // POST /api/v1/signalements — CLIENT : { contratId, contenu }
  create(contratId, contenu) {
    return api.post('/signalements', { contratId, contenu })
  },

  // GET /api/v1/signalements/client?page&size&statut
  getClientSignalements(page = 0, size = 10, statut = null) {
    const params = { page, size }
    if (statut) params.statut = statut
    return api.get('/signalements/client', { params })
  },

  // GET /api/v1/signalements/admin?page&size&statut
  getAll(page = 0, size = 20, statut = null) {
    const params = { page, size }
    if (statut) params.statut = statut
    return api.get('/signalements/admin', { params })
  },

  // GET /api/v1/signalements/{id} — ADMIN
  getById(id) {
    return api.get(`/signalements/${id}`)
  },

  // PUT /api/v1/signalements/{id}/status — ADMIN : { statut, reponseAdmin? }
  updateStatut(id, statut, reponseAdmin = null) {
    return api.put(`/signalements/${id}/status`, { statut, reponseAdmin })
  },

  // PUT /api/v1/signalements/{id}/read — ADMIN
  markAsRead(id) {
    return api.put(`/signalements/${id}/read`)
  },
}
